<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
    <class>MapHubPluginDialogBase</class>
    <widget class="QDialog" name="MapHubPluginDialogBase">
        <property name="geometry">
            <rect>
                <x>0</x>
                <y>0</y>
                <width>600</width>
                <height>500</height>
            </rect>
        </property>
        <property name="windowTitle">
            <string>MapHub: Upload</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
            <item>
                <widget class="QLabel" name="label_layer">
                    <property name="text">
                        <string>Select Layer:</string>
                    </property>
                </widget>
            </item>
            <item>
                <widget class="QComboBox" name="comboBox_layer"/>
            </item>
            <item>
                <widget class="QLabel" name="label_map_name">
                    <property name="text">
                        <string>Map Name:</string>
                    </property>
                </widget>
            </item>
            <item>
                <widget class="QLineEdit" name="lineEdit_map_name"/>
            </item>
            <item>
                <widget class="QLabel" name="label_workspace">
                    <property name="text">
                        <string>Select Workspace:</string>
                    </property>
                </widget>
            </item>
            <item>
                <widget class="QComboBox" name="comboBox_workspace"/>
            </item>
            <item>
                <widget class="QLabel" name="label_folder">
                    <property name="text">
                        <string>Select Destination Folder:</string>
                    </property>
                </widget>
            </item>
            <item>
                <layout class="QHBoxLayout" name="horizontalLayout_folder">
                    <item>
                        <widget class="QPushButton" name="btn_create_folder">
                            <property name="text">
                                <string>Create New Folder</string>
                            </property>
                        </widget>
                    </item>
                </layout>
            </item>
            <item>
                <widget class="QScrollArea" name="scrollArea">
                    <property name="widgetResizable">
                        <bool>true</bool>
                    </property>
                    <widget class="QWidget" name="scrollAreaWidgetContents">
                        <layout class="QVBoxLayout" name="verticalLayout_2">
                            <item>
                                <widget class="QWidget" name="folderContainer" native="true">
                                    <layout class="QVBoxLayout" name="folderLayout">
                                        <!-- Folder items will be added here programmatically -->
                                    </layout>
                                </widget>
                            </item>
                            <item>
                                <spacer name="verticalSpacer_2">
                                    <property name="orientation">
                                        <enum>Qt::Vertical</enum>
                                    </property>
                                    <property name="sizeHint" stdset="0">
                                        <size>
                                            <width>20</width>
                                            <height>40</height>
                                        </size>
                                    </property>
                                </spacer>
                            </item>
                        </layout>
                    </widget>
                </widget>
            </item>
            <item>
                <widget class="QCheckBox" name="checkBox_public">
                    <property name="text">
                        <string>Upload map publicly</string>
                    </property>
                    <property name="checked">
                        <bool>false</bool>
                    </property>
                </widget>
            </item>
            <item>
                <widget class="QDialogButtonBox" name="button_box">
                    <property name="orientation">
                        <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="standardButtons">
                        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
                    </property>
                </widget>
            </item>
        </layout>
    </widget>
    <resources/>
    <connections>
        <connection>
            <sender>button_box</sender>
            <signal>accepted()</signal>
            <receiver>MapHubPluginDialogBase</receiver>
            <slot>accept()</slot>
            <hints>
                <hint type="sourcelabel">
                    <x>20</x>
                    <y>20</y>
                </hint>
                <hint type="destinationlabel">
                    <x>20</x>
                    <y>20</y>
                </hint>
            </hints>
        </connection>
        <connection>
            <sender>button_box</sender>
            <signal>rejected()</signal>
            <receiver>MapHubPluginDialogBase</receiver>
            <slot>reject()</slot>
            <hints>
                <hint type="sourcelabel">
                    <x>20</x>
                    <y>20</y>
                </hint>
                <hint type="destinationlabel">
                    <x>20</x>
                    <y>20</y>
                </hint>
            </hints>
        </connection>
    </connections>
</ui>
